api/src/main/java/org/hawkular/inventory/base/spi/CanonicalPath.java

```diff
@@ -113,6 +114,32 @@ public CanonicalPath visitUnknown(Object entity, Void parameter) {
         }, null);
     }
 
+    public Entity toEntity() {
```

Comment:
Yeah, once my changes for the canonical path are in, I'm actually thinking of using that instead of the full entity references in a lot of places - like relationship targets for example...

We can leave this in for the moment, but I think I will redo this once the resource hierarchy and canonical path changes are in. This will become much simpler.